### DISCUSSION TITLE
geometry source edit mode in output window

### DISCUSSION THREAD

**Discussion by MoorPoor** [brunoherbelin/vimix#123] [Ideas]
Hello,
As I'm currently setting up some installations with the help of Vimix, I noticed in practical use that editing the source geometry in the live output-window would be very useful. actually extremely important for precise mapping on surfaces.
also useful in the case that you cannot connect a control monitor.
as the Geometry distortion and transforming options in F2 view got so powerful recently, this would be a logical next step to add this option in a somewhat reduced way to work on projected output directly.

here is a picture of an an exemplary situation... the vimix window needs to somehow sneak in frame, to position and distort the sources
![photo_2024-02-08_11-25-47](https://github.com/brunoherbelin/vimix/assets/60827766/2df1fa58-3cee-49f2-9be6-9365774ea559)

This obviously goes in the direction of video mapping and hopefully doesn't stray too far from Vimix's original goal?

what you think?

**Comment by brunoherbelin** [comment]
Hey ! Glad to see vimix in action 😄 

So, if I understand well your suggestion, the Display (F5) view correction and management (as below) should be accessible directly in the output window itself ?

![image](https://github.com/brunoherbelin/vimix/assets/60393189/02676f23-2a99-4dc7-b582-ee3775bbea10)

Or is it that you imagine the operations on sources in the Geometry (F2) view should be accessible directly in the output?

**Reply by MoorPoor** [reply]
hello!
yes, more the F2 view options. so each active layer to be manipulable.
I'm glad you're interested :D

**Reply by brunoherbelin** [reply]
Ok now I see.
So I could imagine that the window in F5 could have a big [EDIT] button that would activate the geometry view tools inside the output window directly. Everything is doable, but indeed it could be quite some work...

**Reply by MoorPoor** [reply]
yes, F5 might be the right place for such a button.
I'm sorry, that I cant contribute code-wise

I'm not sure whether 1:1 should be transferred from F2, it's certainly great to have so many parameters, but in live use too many buttons/options can quickly become too confusing. In that case, I'm a fan of additional keyboard shortcuts maybe?
so all the options of transforming from F2 but with a toggle shortcut?

**Comment by MoorPoor** [comment]
I have some more questions and suggestions, but don't know how best to deal with them... I don't want to flood the issues or discussion forum with them. also because I don't know your time management :) what would you suggest?

**Reply by brunoherbelin** [reply]
Discussion here is a good option, but indeed a live chat could be faster :) 
Don't hesitate email me to start a private discussion.

**Reply by MoorPoor** [reply]
I'm currently collecting ideas that we could discuss together. maybe in a public conference? Perhaps one or the other would like to take part? problem with spoken words.... they get lost  :L
